### DISCUSSION TITLE
Poac generate other formats besides Makefile

### DISCUSSION THREAD

**Discussion by Heus-Sueh** [cabinpkg/cabin#990] [Ideas]
One thing that would be interesting to have and would be an attraction of this tool itself, is to generate an output for other build system tools such as: cmake, meson, scons...etc. You can also use Ninja together with these tools if the poac identifies that it is available on the system.

The makefile would be the main build system if no parameters are passed since it is available on almost all systems by default, what do you think of the idea?

**Comment by ken-matsui** [comment]
Hmm, I think generating CMake files would be too costly to maintain. I'm not sure about meson or scons. Generating build.ninja is planned for the future.

**Reply by Heus-Sueh** [reply]
> Hmm, I think generating CMake files would be too costly to maintain. I'm not sure about meson or scons. Generating build.ninja is planned for the future.

Unfortunately cmake has become the standard for many c++ projects and perhaps implementing a generator for it is a necessary evil, I hate it for being too verbose, meson build is a little better and is gaining popularity in several projects, but I don't know if poac would be able to manage more complex projects in both build systems.

**Reply by ken-matsui** [reply]
I still don't see why we want to generate CMake files from Poac. CMake is the standard, why not use it? More specifically, how can Poac be helpful when we generate CMake files (or whatever you've listed)?

**Reply by Heus-Sueh** [reply]
> CMake is the standard, why not use it? More specifically

In my head, poac would be able to translate `poac.toml` to other build systems, but thinking about it now, it's better to have just one build system.
